rust/datafusion/src/logicalplan.rs

```diff
@@ -119,22 +113,24 @@ pub enum ScalarValue {
 }
 
 impl ScalarValue {
-    pub fn get_datatype(&self) -> DataType {
+    pub fn get_datatype(&self) -> Result<DataType> {
```

Comment:
I liked the previous implementation better. Is there a reason why Struct and Null variants should be disallowed? How about List?

Reply:
They just aren't implemented yet ... and I thought Result was better than a kernel panic. Maybe we just leave this one as a panic?